pkg/registry/pod/rest.go

```diff
@@ -55,6 +55,9 @@ func (podStrategy) PrepareForCreate(obj runtime.Object) {
 	pod.Status = api.PodStatus{
 		Phase: api.PodPending,
 	}
+	for i := range pod.Spec.ImagePullSecrets {
+		pod.Spec.ImagePullSecrets[i] = api.ObjectReference{Name: pod.Spec.ImagePullSecrets[i].Name}
```

Comment:
@lavalamp The idea was to use ObjectReferences for now and refactor later to a narrow object reference-like thing that has only the name field.